src/Shared/ApiExplorerTypes/ProducesResponseTypeMetadata.cs

```diff
@@ -65,30 +62,29 @@ public ProducesResponseTypeMetadata(Type type, int statusCode, string contentTyp
         _contentTypes = GetContentTypes(contentType, additionalContentTypes);
     }
 
+    // Only for internal use where validation is unnecessary.
+    private ProducesResponseTypeMetadata(Type? type, int statusCode, IEnumerable<string> contentTypes)
+    {
+
+        Type = type;
+        StatusCode = statusCode;
+        _contentTypes = contentTypes;
+    }
+
     /// <summary>
     /// Gets or sets the type of the value returned by an action.
     /// </summary>
-    public Type Type { get; set; }
+    public Type? Type { get; set; }
```

Comment:
Not sure making this nullable is the right call here. Can you clarify your reasoning as to why this is?

Reply:
This type is internal. The interface is public, and `Type` is already nullable there.  Everything reading `Type` is using the interface is dealing with the possible null anyway for custom interface implementations which is why I didn't need to react.